src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -8879,6 +8892,11 @@ private ArgumentSyntax ParseArgumentExpression(bool isIndexer)
                 expression = this.ParseSubExpression(0);
             }
 
+            if (refOrOutKeyword != null && IsLambda(expression.Kind))
+            {
+                expression = this.AddError(expression, ErrorCode.ERR_RefLvalueExpected);
```

Comment:
Does this actually violate language grammar? Otherwise, it feels appropriate to report this error during binding instead.